command/agent.go

```diff
@@ -614,3 +653,34 @@ func (c *AgentCommand) removePidFile(pidPath string) error {
 	}
 	return os.Remove(pidPath)
 }
+
+// setupTemplateConfig creates a config.Vault struct for use by Consul Template.
+// Consul Template does not currently allow us to pass in a configured API
+// client, unlike the AuthHandler and SinkServer that reuse the client creted in
+// this Run() method. Here we build a config.Vault struct for use by the
+// Template Server that matches the configuration used to create the client
+// (c.client), but in a struct of type config.Vault so that Consul Template can
+// create it's own api client internally.
+// TODO test setupTemplateConfig
+func (c *AgentCommand) setupTemplateConfig() *config.Vault {
+	cfg := new(config.Vault)
```

Comment:
I think since you're just checking for the zero value, all of this can be a single literal:

```
return &config.Vault {
    Address:   c.flagAddress,
    CACert:    c.flagCACert,
    ...
}